driver-core/src/main/com/mongodb/connection/ClusterDescription.java

```diff
@@ -186,6 +208,16 @@ public ClusterType getType() {
         return type;
     }
 
+    /**
+     * Gets any exception encountered while resolving the SRV record for the initial host.
+     *
+     * @return any exception encountered while resolving the SRV record for the initial host, or null if none
+     * @since 3.10
+     */
+    public MongoConfigurationException getSrvResolutionException() {
```

Comment:
This type may prove too inflexible.  Maybe it should be MongoException.

Reply:
Agreed